lib/src/rewrite.rs

```diff
@@ -830,3 +830,150 @@ pub fn move_commits(
         num_skipped_rebases,
     })
 }
+
+pub struct CommitToSquash {
+    commit: Commit,
+    selected_tree: MergedTree,
+    parent_tree: MergedTree,
+}
+
+impl CommitToSquash {
+    pub fn new(repo: &dyn Repo, commit: Commit, selected_tree: MergedTree) -> BackendResult<Self> {
+        let parent_tree = commit.parent_tree(repo)?;
```

Comment:
nit: `parent_tree()` might have to merge trees. Maybe it's okay to just make all `CommitToSquash` fields public and let callers fill out?